src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -104,641 +105,695 @@
+            BinaryOperator<String> reducer = new BinaryOperator<String>() {
+                private boolean notificationFound = false;
+
+                @Override
+                public String apply(String s, String s2) {
+                    if (s2.startsWith("---")) {
+                        notificationFound = true;
+                    }
+                    return notificationFound ? s : s + s2 + "\n";
+                }
+            };
+            return reader.lines().reduce("", reducer);
+        }
+    }
+
+    private static String execute(String command, Map<String, String> envs, String... args) throws IOException {
+        return execute(new File(HOME_FOLDER), command, envs, args);
+    }
+
+    @Override
+    public void describeApplication(String project, String application) throws IOException {
+        ExecHelper.executeWithTerminal(envVars, command, "app", "describe", application, "--project", project);
+    }
+
+    @Override
+    public void deleteApplication(String project, String application) throws IOException {
+        execute(command, envVars, "app", "delete", application, "-f", "--project", project);
+    }
+
+    @Override
+    public void push(String project, String application, String context, String component) throws IOException {
+        ExecHelper.executeWithTerminal(new File(context), envVars, command, "push");
+    }
+
+    @Override
+    public void describeComponent(String project, String application, String context, String component) throws IOException {
+        if (context != null) {
+            ExecHelper.executeWithTerminal(new File(context), envVars, command, "describe");
+        } else {
+            ExecHelper.executeWithTerminal(envVars, command, "describe", "--project", project, "--app", application, component);
+        }
+    }
+
+    @Override
+    public void watch(String project, String application, String context, String component) throws IOException {
+        ExecHelper.executeWithTerminal(new File(context), envVars, command, "watch");
+    }
+
+    @Override
+    public void createComponentLocal(String project, String application, String componentType, String componentVersion, String component, String source, boolean push) throws IOException {
+        List<String> args = new ArrayList<>();
+        args.add(command);
+        args.add("create");
+        if (StringUtils.isNotBlank(componentVersion)) {
+            args.add(componentType + ":" + componentVersion);
+        } else {
+            args.add(componentType);
+        }
+        args.add(component);
+        args.add("--project");
+        args.add(project);
+        args.add("--app");
+        args.add(application);
+        if (StringUtils.isNotBlank(componentVersion)) {
+            args.add("--s2i");
+        }
+        if (push) {
+            args.add("--now");
+        }
+        ExecHelper.executeWithTerminal(new File(source), args.toArray(new String[0]));
```

Comment:
this is wrong: you need tp pass envVars to support proxy

Reply:
fixed, and also removed method without envVars, not used anymore.